src/storage-client/src/storage_collections.rs

```diff
@@ -137,166 +111,10 @@ pub trait StorageCollections: Debug {
     /// an error if the collection does not exist.
     fn check_exists(&self, id: GlobalId) -> Result<(), StorageError<Self::Timestamp>>;
 
-    /// Returns aggregate statistics about the contents of the local input named
-    /// `id` at `as_of`.
-    async fn snapshot_stats(
-        &self,
-        id: GlobalId,
-        as_of: Antichain<Self::Timestamp>,
-    ) -> Result<SnapshotStats, StorageError<Self::Timestamp>>;
-
-    /// Returns aggregate statistics about the contents of the local input named
-    /// `id` at `as_of`.
-    ///
-    /// Note that this async function itself returns a future. We may
-    /// need to block on the stats being available, but don't want to hold a reference
-    /// to the controller for too long... so the outer future holds a reference to the
-    /// controller but returns quickly, and the inner future is slow but does not
-    /// reference the controller.
-    async fn snapshot_parts_stats(
-        &self,
-        id: GlobalId,
-        as_of: Antichain<Self::Timestamp>,
-    ) -> BoxFuture<'static, Result<SnapshotPartsStats, StorageError<Self::Timestamp>>>;
-
-    /// Update the given [`StorageTxn`] with the appropriate metadata given the
-    /// IDs to add and drop.
-    ///
-    /// The data modified in the `StorageTxn` must be made available in all
-    /// subsequent calls that require [`StorageMetadata`] as a parameter.
-    async fn prepare_state(
-        &self,
-        txn: &mut (dyn StorageTxn<Self::Timestamp> + Send),
-        ids_to_add: BTreeSet<GlobalId>,
-        ids_to_drop: BTreeSet<GlobalId>,
-    ) -> Result<(), StorageError<Self::Timestamp>>;
-
-    /// Create the collections described by the individual
-    /// [CollectionDescriptions](CollectionDescription).
-    ///
-    /// Each command carries the source id, the source description, and any
-    /// associated metadata needed to ingest the particular source.
-    ///
-    /// This command installs collection state for the indicated sources, and
-    /// they are now valid to use in queries at times beyond the initial `since`
-    /// frontiers. Each collection also acquires a read capability at this
-    /// frontier, which will need to be repeatedly downgraded with
-    /// `allow_compaction()` to permit compaction.
-    ///
-    /// This method is NOT idempotent; It can fail between processing of
-    /// different collections and leave the [StorageCollections] in an
-    /// inconsistent state. It is almost always wrong to do anything but abort
-    /// the process on `Err`.
-    ///
-    /// The `register_ts` is used as the initial timestamp that tables are
-    /// available for reads. (We might later give non-tables the same treatment,
-    /// but hold off on that initially.) Callers must provide a Some if any of
-    /// the collections is a table. A None may be given if none of the
-    /// collections are a table (i.e. all materialized views, sources, etc).
-    async fn create_collections(
```

Comment:
IMO `create_collections` also has to be on the collections interface. Though, yes, these ones are a bit weird because they're really more a `hey_this_collection_has_been_created_so_set_up_your_state`. A pure "consumer" of `StorageCollections` wouldn't need this, but none of our controller-like traits have a split between pure users and parts that actually change things.